packages/amplication-server/src/core/resource/resourceBtm.service.ts

```diff
@@ -211,29 +220,37 @@ export class ResourceBtmService {
     const usedDuplicatedEntities = new Set<string>();
 
     const originalResource = await this.getResourceDataForBtm(resourceId);
-    const originalResourceEntitiesSet = new Set(
+    const originalResourceEntityNamesSet = new Set(
       originalResource.entities.map((entity) => entity.name)
     );
 
+    const inventedEntitiesByGpt = recommendedResourceEntities.filter(
+      (item) => !originalResourceEntityNamesSet.has(item)
+    );
+
     return {
       microservices: promptResultObj.microservices
-        .sort((microservice) => -1 * microservice.tables.length)
+        .sort((a, b) => a.tables.length - b.tables.length)
         .map((microservice) => ({
           name: microservice.name,
           functionality: microservice.functionality,
           tables: microservice.tables
-            .filter((dataModelName) => {
+            .filter((tableName) => !inventedEntitiesByGpt.includes(tableName))
+            .filter((tableName) => {
```

Comment:
no need for 2 filters right? you should be able to do everything in a single run